airbyte-integrations/connectors/source-singlestore/src/main/java/io/airbyte/integrations/source/singlestore/SinglestoreSource.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.singlestore;
+
+import static io.airbyte.cdk.db.jdbc.JdbcUtils.EQUALS;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.CLIENT_KEY_STORE_PASS;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.CLIENT_KEY_STORE_TYPE;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.CLIENT_KEY_STORE_URL;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.TRUST_KEY_STORE_PASS;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.TRUST_KEY_STORE_TYPE;
+import static io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils.TRUST_KEY_STORE_URL;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.cdk.db.factory.DatabaseDriver;
+import io.airbyte.cdk.db.jdbc.JdbcUtils;
+import io.airbyte.cdk.db.jdbc.streaming.AdaptiveStreamingQueryConfig;
+import io.airbyte.cdk.integrations.base.IntegrationRunner;
+import io.airbyte.cdk.integrations.base.Source;
+import io.airbyte.cdk.integrations.source.jdbc.AbstractJdbcSource;
+import io.airbyte.cdk.integrations.source.jdbc.JdbcSSLConnectionUtils;
+import io.airbyte.commons.json.Jsons;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.sql.JDBCType;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SinglestoreSource extends AbstractJdbcSource<JDBCType> implements Source {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SinglestoreSource.class);
+  public static final String DRIVER_CLASS = DatabaseDriver.SINGLESTORE.driverClassName;
+
+  public SinglestoreSource() {
+    super(DRIVER_CLASS, AdaptiveStreamingQueryConfig::new, JdbcUtils.defaultSourceOperations);
+  }
+
+  @Override
+  public JsonNode toDatabaseConfig(final JsonNode config) {
+    final String encodedDatabaseName = URLEncoder.encode(
+        config.get(JdbcUtils.DATABASE_KEY).asText(), StandardCharsets.UTF_8);
+    final StringBuilder jdbcUrl = new StringBuilder(
+        String.format("jdbc:singlestore://%s:%s/%s", config.get(JdbcUtils.HOST_KEY).asText(),
+            config.get(JdbcUtils.PORT_KEY).asText(), encodedDatabaseName));
+    // ensure the return year value is a Date; see the rationale
+    jdbcUrl.append("?yearIsDateType=false");
```

Comment:
Here we set that YEAR is not a DATE type.
Comment tells the opposite of what is done in the code

Reply:
Removing comment, year type will be used as integer